### DISCUSSION TITLE
Allow to run RvLoader from SD card

### DISCUSSION THREAD

**Discussion by creedof69** [Aurelio92/RVLoader#40] [Ideas]
Hello I'm here to submit another idea.
I know if there is already a way that works why would you spend time to do this but, that would be great if We could choose to run Rvloader from the sd card just like other loader. That would permit us (me, but I imagine I'm not the only one) to concentrate all file in the same device. Games in usb drive, Homebrews in sd card. For exemple I had to change the directory of covers for usbloadergx if I would rvloader displayed covers without doubled them.
I hope you will think about it. Thanks for reading

**Comment by erictronic7** [comment]
I'm working on implementing code that allows the use of SD memories. I have practically set everything up to load GC games and HBC. To start VC and Wii games, it's necessary to incorporate the configuration into Hiidra, but its source code is not yet available, 
I hope it can be published soon.

**Reply by Aurelio92** [reply]
Do you have a fork I can look at? Hiidra's source will be released soon-ish most likely

**Reply by erictronic7** [reply]
Of course, I'm not really a software developer. I have some knowledge of C and CPP. I used AI to understand the base code and integrated the sdhc and wiisd_io libraries to mount the SD instead of the USB. I replaced the directories for covers, themes, etc. with those on the SD. So far, it works perfectly in the tests. In order to load Wii and VC games, I need to have a clone of the SD data on the USB because Hiidra directs itself to the USB upon startup.
[RVLoader-1.6_SDHC.zip](https://github.com/Aurelio92/RVLoader/files/11536710/RVLoader-1.6_SDHC.zip)

**Reply by creedof69** [reply]
I don't know how your code is, if you strictly take back the Aurelio one or not, but the "must have" would be to be able to set folder paths in the loader settings. Like this, anyone could run Rvloader from device they want, choose a custom location for Nintendont, Cover and others...

**Reply by erictronic7** [reply]
For now, just for testing purposes, I replaced the USB directories with the SD directories. However, my intention is to establish the device selection from the bootloader, so we can choose from where we want our RVLoader to start.

**Reply by DraconicNEO** [reply]
Are you still working on this by chance @erictronic7

**Comment by inukaze** [comment]
Someone is still develop this feature ? i am using microsd class 10 of 32GB, but i just had use 4GB xD.

**Comment by SuperMarioEpicTeam** [comment]
Hello, 'call me Walid!

(I am new in commenting in GitHub, so apologies in advance if what follows is off-topic. Nonetheless please take it into consideration)

I need a few portions of the USB Loader GX code to be ported to my mod of a Wii game. However, the app's code calls functions which aren't defined anywhere (especially GD functions).

I need their definition. Thank you in advance!

**Reply by Aurelio92** [reply]
This is not USB Loader GX repository
